src/main/kotlin/anttracker/Release.kt

```diff
@@ -23,33 +26,115 @@ value class ReleaseId(
             "Release id length must be between 1 and 8 characters"
         }
     }
+
+    override fun toString(): String = id
 }
 
-class Release(
-    val releaseName: ReleaseId,
-    val product: Product,
-    var releaseDate: LocalDate,
-)
+val promptEnterRel = "\nPlease enter new release name. ` to abort:"
+
+val promptSelectRel = "\nPlease select release. ` to abort:"
+
+val promptSelectAffRel = "\nPlease select affected release. ` to abort:"
 
 // -------------------------------------------------------------------------------
 // Prints a sub-menu of selecting a product to create a release for.
 // Use by including in main-menu loop; should be triggered upon user selecting
 // New Release from the main menu. Thus, upon function return, system returns to main menu.
 // ---
 fun menu() {
-    TODO()
+    println("== NEW RELEASE ==")
+    // val selectedProduct = selectProduct()    <- TODO: uncomment and replace when implemented in product module
+    val selectedProduct = "Product 1"
+    when (val selection = readln()) {
+        "`" -> return
+        "1" -> createRelease(selectedProduct)
+        else -> {
+            println("Bad input: $selection.")
+            return
+        }
+    }
 }
 
 // -------------------------------------------------------------------------------
-// Prints to console a paginated list of releases for a product
-// Returns a string indicating user input:
-//  "`": user exit
-//  an int (e.g. "4"): a line number selecting a particular release to query on
-// Call as part of any relevant sub-menu
-//  e.g. During request creation, call this to find and select the affected release
+// A read-only display of releases for a product. Returns control upon reaching
+//  the final page, or if user aborts with backtick.
 // ---
-fun displayReleases(
-    product: String, // in
-): String {
-    TODO()
+fun displayReleases(productName: String) {
+    val relPage = PageOfReleases(productName)
+    relPage.loadRecords()
+    relPage.display()
+
+    while (!relPage.lastPage()) {
+        val userInput = readln()
+        when (userInput) {
+            "`" -> return
+            "" -> {
+                relPage.loadNextPage()
+                relPage.display()
+            }
+        }
+    }
+    return
+}
+
+fun selectRelease(productName: String): Release? {
+    val relPage = PageOfReleases(productName)
+    relPage.loadRecords()
+    relPage.display()
+
+    var linenum: Int? = null
+    while (linenum == null) {
+        println(promptSelectRel) // "Please select release. ` to abort: "
+        val userInput = readln()
+        when (userInput) {
+            "`" -> return null
+            "" -> {
```

Comment:
Could you add some comments before each of these cases, saying "user indicates intention to leave early" (`), "user indicates for next page" (enter), and "user possibly selects a row" (else)

Reply:
done